### ISSUE TITLE
How Do You Evenly Space a Known Number of Tabs?

### ISSUE THREAD

**Issue by maxmagee** [Slowyn/react-native-underline-tabbar#32]
Say I have a set number of tabs and I know the labels won't overflow the width of the screen.

I'm struggling with how to evenly space the labels across the header like the below example:

![snip20180810_16](https://user-images.githubusercontent.com/9301824/43987676-3efa6cd4-9cea-11e8-9b82-5341606c76d7.png)

Basically I want to use 'space-around' as if they were flex items but I can't seem to get it to work properly.

Any help would be greatly appreciated, thanks!

**Comment by Slowyn** [comment]
@maxmagee
Hi Max! You can achieve such behavior passing through some styles to component. Take a look:
```js
        <ScrollableTabView
          tabBarUnderlineColor="#53ac49"
          tabBarActiveTextColor="#53ac49"
          renderTabBar={() => (
            <TabBar
              scrollContainerStyle={{
                width: '100%',
                justifyContent: 'space-between',
              }}
            />
        )}
        >
          <Page tabLabel={{label: "Page #1"}} label="Page #1"/>
          <Page tabLabel={{label: "Page #2", badge: 3}} label="Page #2 aka Long!"/>
          <Page tabLabel={{label: "Page #3"}} label="Page #3"/>
        </ScrollableTabView>
```
The result:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/7250604/43996005-825cf6c6-9dc2-11e8-8d99-1166c0fb4269.png">

Also while I was checking this, I have found out the that when I apply 'space-around', calculations for underline become incorrect. So be careful with this.
This is some kind of undesigned behavior because the initial purpose of library is tabs that don't fit into a container. I hope everything will work fine for you, but if not feel free to report an issue.
P.S. Probably you can achieve the same result with inbuilt TabBars of ReactNativeScollableTabView.

**Comment by Slowyn** [comment]
And just for your information. There are a lot of opportunities for styling of every part of tab. Also you can pass you own view to tabbar if you need custom animations or a different layout. I see now how documentation doesn't give you enough information.

**Comment by maxmagee** [comment]
Hey, thanks for replying and with the heads up regarding space-between. I'm totally in agreement that this isn't within the scope of this library after thinking more about it (it's designed for like you said, tabs that don't fit within a staticly-sized container).

I tried playing around with the space-between, but it's not working great for me with two tabs and paddingHorizontal on the scroll-container only affects the left side (so it doesn't bring both tabs toward the middle). Again, that's kind of expected since it's a scrollView.

That beings said, I did find a hack-y work-around that works for two tabs and _should_ work for more than 2 with a caveat that you need to know the width of the tabs when rendered beforehand.

If anyone runs across this in the future and wants to know how I did it, below is what I did (tab names generalized for this example):

```
const TAB_A_WIDTH = 34;
const TAB_B_WIDTH = 56;
const ALL_TABS_WIDTH = TAB_A_WIDTH + TAB_B_WIDTH;
const TOTAL_NUMBER_OF_TABS = 2;
const TOTAL_WIDTH = Dimensions.get('window').width;
const TAB_GAP_WIDTH = (TOTAL_WIDTH - ALL_TABS_WIDTH) / (TOTAL_NUMBER_OF_TABS + 1);
```

Then I set the tabMargin property of the TabBar component to be TAB_GAP_WIDTH.

![snip20180811_18](https://user-images.githubusercontent.com/9301824/43996161-88029576-9d82-11e8-9c7a-ec2bfc1c43ef.png)

Basically I just used math to find out how much 'gap space' I had to distribute between the tabs. The major downside to this approach is you have to adjust the numbers if the text or font changes.

Thanks again for your library and your reply. I'll consider this closed.

Cheers!
